modules/system-view/src/main/java/org/apache/ignite/internal/systemview/SystemViewManagerImpl.java

```diff
@@ -0,0 +1,168 @@
+
+    /** Prevents double stopping of the component. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
+    /** Future which is completed when system views are registered in the catalog. */
+    private final CompletableFuture<Void> viewsRegistrationFuture = new CompletableFuture<>();
+
+    /** Creates a system view manager. */
+    public SystemViewManagerImpl(CatalogManager catalogManager) {
+        this.catalogManager = catalogManager;
+    }
+
+    @Override
+    public void start() {
+        inBusyLock(busyLock, () -> {
+            if (!startGuard.compareAndSet(false, true)) {
+                throw new IllegalStateException("System view manager cannot be started twice");
+            }
+
+            if (views.isEmpty()) {
+                viewsRegistrationFuture.complete(null);
+
+                return;
+            }
+
+            catalogManager.execute(prepareCatalogCommands(views.values())).whenComplete(
+                    (r, t) -> viewsRegistrationFuture.complete(null)
+            );
+
+            nodeAttributes.put(NODE_ATTRIBUTES_KEY, String.join(NODE_ATTRIBUTES_LIST_SEPARATOR, views.keySet()));
+        });
+    }
+
+    @Override
+    public void stop() throws Exception {
+        if (!stopGuard.compareAndSet(false, true)) {
+            return;
+        }
+
+        viewsRegistrationFuture.completeExceptionally(new NodeStoppingException());
+
+        busyLock.block();
+    }
+
+
+    /** {@inheritDoc} */
+    @Override
+    public void register(SystemView<?> view) {
+        if (views.containsKey(view.name())) {
+            throw new IllegalArgumentException(format("The view with name '{}' already registered", view.name()));
+        }
+
+        inBusyLock(busyLock, () -> {
+            if (startGuard.get()) {
+                throw new IllegalStateException(format("Unable to register view '{}', manager already started", view.name()));
+            }
+
+            views.put(view.name(), view);
+        });
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Map<String, String> nodeAttributes() {
+        return nodeAttributes;
+    }
+
+    /**
+     * Returns future which is completed when system views are registered in the catalog.
+     */
+    public CompletableFuture<Void> completeRegistration() {
+        return viewsRegistrationFuture;
+    }
+
+    private List<CatalogCommand> prepareCatalogCommands(Collection<SystemView<?>> views) {
+        List<CatalogCommand> catalogCommands = new ArrayList<>(views.size());
+
+        for (SystemView<?> view : views) {
+            catalogCommands.add(prepareCatalogCommand(view));
+        }
+
+        return catalogCommands;
+    }
+
+    private CatalogCommand prepareCatalogCommand(SystemView<?> view) {
+        List<ColumnParams> columnParams = new ArrayList<>(view.columns().size());
+
+        for (SystemViewColumn<?, ?> col : view.columns()) {
+            columnParams.add(prepareColumnParams(col));
```

Comment:
I think we need to add an extra column of type `String`  for `nodeNameColumnAlias` in case of node system view.

Reply:
Column added, thanks.